bindings/profilers/heap.cc

```diff
@@ -521,6 +554,17 @@ NAN_METHOD(HeapProfiler::StopSamplingHeapProfiler) {
   auto isolate = info.GetIsolate();
   isolate->GetHeapProfiler()->StopSamplingHeapProfiler();
   PerIsolateData::For(isolate)->GetHeapProfilerState().reset();
+
+  // Remove cleanup hook since profiler is explicitly stopped
+  {
+    const std::lock_guard<std::mutex> lock(g_heap_profiler_mutex);
+    if (g_heap_profiler_isolates.find(isolate) !=
+        g_heap_profiler_isolates.end()) {
+      node::RemoveEnvironmentCleanupHook(
+          isolate, HeapProfilerCleanupHook, isolate);
+      g_heap_profiler_isolates.erase(isolate);
+    }
```

Comment:
You can simplify this by speculatively invoking `erase()` and checking its return value.

```suggestion
    if (g_heap_profiler_isolates.erase(isolate) == 1) {
      node::RemoveEnvironmentCleanupHook(
          isolate, HeapProfilerCleanupHook, isolate);
    }
```